packages/form-core/src/FieldApi.ts

```diff
@@ -332,10 +332,15 @@ export class FieldApi<
   validateSync = (value = this.state.value, cause: ValidationCause) => {
     const { onChange, onBlur } = this.options
     const validate =
-      cause === 'submit' ? undefined : cause === 'change' ? onChange : onBlur
+      cause === 'submit'
+        ? onChange
+          ? onChange
+          : onBlur
```

Comment:
This is not the correct backup to `cause === "submit" && !onChange`

Instead, we need to run both `onChange` **and** `onBlur`.

I'll take over this PR to make the appropriate changes.